### PR TITLE
Check ssh connection

### PR DESCRIPTION
**Pull request by DAGpro**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #158

### CODE REVIEW

File: src/App/Component/Console/PackageCommand.php [hunk 1]
```diff
@@ -346,4 +347,23 @@ protected function getExampleCommandPrefix(): string
...
     }
+
+    protected function checkSSHConnection(): void
+    {
+        $process = new Process(['ssh', '-T', 'git@github.com']);
+        $process
+            ->setTimeout(null)
+            ->run();
+
+        if ($process->getExitCode() !== 1) {
```
Comment (samdark) [inline src/App/Component/Console/PackageCommand.php:358]: Shouldn't we check for `0` as a success return code?

File: src/App/Component/Console/PackageCommand.php [hunk 2]
```diff
@@ -346,4 +347,23 @@ protected function getExampleCommandPrefix(): string
...
+            $this
+                ->getIO()
+                ->error([
+                    'Checking access to github.com ... DENIED',
```
Comment (samdark) [inline src/App/Component/Console/PackageCommand.php:362]: Can't we take the error from the `ssh` itself?

Comment (DAGpro) [reply src/App/Component/Console/PackageCommand.php:362]: display a more informative message

As mentioned in the issue, the message is confusing

Comment (samdark) [reply src/App/Component/Console/PackageCommand.php:362]: I mean it could be another reason, not only denied access.

Comment (DAGpro) [reply src/App/Component/Console/PackageCommand.php:362]: added errors.

Is it correct to end the command with the exit construct?

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks.

**Pull request merged**